### ISSUE TITLE
Cannot connect to Ryuk on Ubuntu 18.04 and docker 18.09

### ISSUE THREAD

**Issue by vusalalishov** [testcontainers/testcontainers-java#1418]
I'm getting the next exception when trying to create a MySQL container:
```
java.lang.ExceptionInInitializerError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:217)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.createTest(SpringJUnit4ClassRunner.java:226)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner$1.runReflectiveCall(SpringJUnit4ClassRunner.java:289)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.methodBlock(SpringJUnit4ClassRunner.java:291)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:246)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.IllegalStateException: Can not connect to Ryuk
	at org.testcontainers.utility.ResourceReaper.start(ResourceReaper.java:148)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:125)
	at org.testcontainers.containers.GenericContainer.<init>(GenericContainer.java:138)
	at org.testcontainers.containers.JdbcDatabaseContainer.<init>(JdbcDatabaseContainer.java:45)
	at org.testcontainers.containers.MySQLContainer.<init>(MySQLContainer.java:29)
	at com.xxx.yyy.BaseIntegrationTest.<clinit>(BaseIntegrationTest.java:25)
	... 25 more
```
The `docker ps` shows ryuk container is running but it doesn't respond to any requests to it.
Also, I have tried to run `moby-ryuk` on my host machine and it works (I'm getting the `ACK` response to my requests). It looks like there are problems when running it (`moby-ryuk`) in the docker container.

I use testcontainers 1.11.1

**Comment by kiview** [comment]
Hi @vusalalishov 

>  It looks like there are problems when running it (moby-ryuk) in the docker container.

Does this mean you are running some kind of Docker-in-Docker setup? Also how did you install Docker? Using the official Docker-CE repo, Docker package in Ubutnu or snap package?

**Comment by vusalalishov** [comment]
> Does this mean you are running some kind of Docker-in-Docker setup

No, I have a host machine running ubuntu and I have installed docker ce on it.
> Using the official Docker-CE repo, Docker package in Ubutnu or snap package?

I didn't use snap, I followed the next article for the installation: https://docs.docker.com/install/linux/docker-ce/ubuntu/

```
$ docker version
Client:
 Version:           18.09.5
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        e8ff056
 Built:             Thu Apr 11 04:44:15 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.5
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       e8ff056
  Built:            Thu Apr 11 04:10:53 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by barlog-m** [comment]
Have some issue with last Arch Linux. Testcontainers works once after computer reboot and next runs returns this exception.
Docker 18.09.6-ce, Linux 5.1.3

**Comment by collinpeters** [comment]
Found this issue as I'm running Arch and saw the previous comment on Arch. Will put my results here for anyone else who finds this. Today I switched Internet providers and the new router was running on 10.0.0.x (still 255.255.255.0 netmask) instead of the previous 192.168.0.x. Somehow this messed up something and I got the Ryuk error. I simply switch the network to 192.168.1.x and it started working again. I assume it was a Docker issue treating a 10. network differently somehow. I didn't look further as I was happy to have it running. Hope this helps someone.
